### PR TITLE
Fix put permissions (fixes #661)

### LINKED ISSUE #661: data lost on object with PUT to replace permissions
```
echo '{"data": {"tata": 42}}' | http PUT :8888/v1/buckets/garde --auth user:pass
HTTP/1.1 201 Created
Access-Control-Expose-Headers: Retry-After, Content-Length, Alert, Backoff
Content-Length: 166
Content-Type: application/json; charset=UTF-8
Date: Mon, 30 May 2016 10:04:51 GMT
Etag: "1464602691600"
Last-Modified: Mon, 30 May 2016 10:04:51 GMT
Server: waitress

{
    "data": {
        "id": "garde", 
        "last_modified": 1464602691600, 
        "tata": 42
    }, 
    "permissions": {
        "write": [
            "basicauth:f505765817a6b4ea46278be0620ddedd83b10f71f7695683719fe001cf0871d7"
        ]
    }
}

```

```
echo '{"permissions": {"write": ["bizbiz"]}}' | http PUT :8888/v1/buckets/garde --auth user:pass
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Retry-After, Content-Length, Alert, Backoff
Content-Length: 216
Content-Type: application/json; charset=UTF-8
Date: Mon, 30 May 2016 10:05:54 GMT
Etag: "1464602754354"
Last-Modified: Mon, 30 May 2016 10:05:54 GMT
Server: waitress

{
    "data": {
        "id": "garde", 
        "last_modified": 1464602754354
    }, 
    "permissions": {
        "collection:create": [], 
        "group:create": [], 
        "read": [], 
        "write": [
            "bizbiz", 
            "basicauth:f505765817a6b4ea46278be0620ddedd83b10f71f7695683719fe001cf0871d7"
        ]
    }
}
```
Interesting to note that the contrary is not true:
```
echo '{"data": {"tata": 42}}' | http PUT :8888/v1/buckets/garde --auth user:pass
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Retry-After, Content-Length, Alert, Backoff
Content-Length: 226
Content-Type: application/json; charset=UTF-8
Date: Mon, 30 May 2016 10:06:00 GMT
Etag: "1464602760183"
Last-Modified: Mon, 30 May 2016 10:06:00 GMT
Server: waitress

{
    "data": {
        "id": "garde", 
        "last_modified": 1464602760183, 
        "tata": 42
    }, 
    "permissions": {
        "collection:create": [], 
        "group:create": [], 
        "read": [], 
        "write": [
            "bizbiz", 
            "basicauth:f505765817a6b4ea46278be0620ddedd83b10f71f7695683719fe001cf0871d7"
        ]
    }
}

```

### PR DESCRIPTION
**Pull request by leplatrem**
This fix only works with schemaless resources (groups, collections, records).

On resource with schema (eg. group with its `members` attribute), trying to replace the permissions with a `PUT` without `data` will result in a 400 response.

As explained in code comments, this cannot be done better without cleaning the way Cornice manages Colander schemas.

At least, this prevents the data from being implicitly lost, which is a good first step IMO

@almet || @Natim r?

### CODE REVIEW

File: kinto/tests/test_views_groups.py [hunk 1]
```diff
@@ -187,16 +187,18 @@ def setUp(self):
         super(InvalidGroupTest, self).setUp()
         self.create_bucket('beers')
 
-    def test_groups_must_have_members_attribute(self):
+    def test_groups_data_is_required(self):
```
Comment (Natim) [inline kinto/tests/test_views_groups.py]: This means that we cannot change group permissions if we do not provide also the data with PUT.
But is it working fine with PATCH?

File: kinto/tests/core/resource/test_views.py [hunk 2]
```diff
@@ -82,6 +82,39 @@ def test_permissions_are_not_modified_if_not_specified(self):
...
+        body = {'data': MINIMALIST_RECORD,
+                'permissions': {'read': ['group:readers']}}
+        resp = self.app.post_json(self.collection_url, body,
+                                  headers=self.headers)
+        object_uri = self.get_item_url(resp.json['data']['id'])
+
+        body.pop('data')
+        # With the current Cornice limitations, it is not possible to define
+        # a validator that makes sure that at least of `data` or `permissions`
+        # is specified in body.
```
Comment (almet) [inline kinto/tests/core/resource/test_views.py]: Can you link to the cornice issue in the docs?

**Comment by almet** [comment]
This looks good to me!

**Comment by Natim** [comment]
LGTM

**Pull request merged**